### PR TITLE
Fixing BigtableOptions

### PR DESCRIPTION
**Pull request by sduskis**
PR #1736 shows that the current BigtableOptions code is brittle.  useCachedDataPool was not being transfered in the manual cloning process.  This PR is a proposal for a fix, and also a request for proposlas.

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java [hunk 1]
```diff
@@ -544,4 +469,12 @@ public Builder toBuilder() {
   public boolean useCachedChannel() {
     return useCachedDataPool;
   }
+
+  protected BigtableOptions clone() {
+    try {
+      return (BigtableOptions) super.clone();
```
Comment (spollapally) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java]: Wouldn't this inadvertently update the reference type fields already in use before the clone, and effectively make them mutable?

**Comment by igorbernstein2** [comment]
Have you thought about using auto value?

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java (hunk 1 above)
Comment (sduskis) [reply bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java]: You mean objects like `CallOptionsConfig` and `RetryOptions`?  Those are supposed to be immutable.

**Comment by sduskis** [comment]
@igorbernstein2, yup, I added a comment to that affect.

**Comment by spollapally** [comment]
LGTM

**Pull request merged**